### PR TITLE
Move from drone to Github Actions

### PR DESCRIPTION
**Pull request by mitulshah-suse**
Issue: https://github.com/rancher/rancher/issues/45376

### CODE REVIEW

File: scripts/ci
Comment (HarrisonWAffel) [inline scripts/ci:4]: Using `$HOME` here will prevent us from building this repository locally, which probably isn't a good thing. I can't speak on how often (if ever) this repo is built locally, so I'll leave it up to the other reviewers to determine if this is acceptable or not - just want to point it out

File: .github/workflows/workflow.yaml
Comment (HarrisonWAffel) [inline .github/workflows/workflow.yaml:25]: Just to check my understanding, is this step needed because we are using the dapper image for this job?

File: .github/workflows/workflow.yaml
Comment (HarrisonWAffel) [inline .github/workflows/workflow.yaml:36]: Again I think breaking these steps up into separate files would be easier to maintain, but this is more of a nit

File: scripts/ci
Comment (mitulshah-suse) [reply scripts/ci:4]: `dirname $0` resolves to `.` on these runners, which is why i had to change it.

File: .github/workflows/workflow.yaml
Comment (mitulshah-suse) [reply .github/workflows/workflow.yaml:25]: No, its needed because we are running git commands as part of the `version` script. Without this step the `.git` folder does not get downloaded on the runner and all `git` commands start failing with not a git repository error.
ref: https://github.com/actions/checkout/issues/363

File: .github/workflows/workflow.yaml
Comment (mitulshah-suse) [reply .github/workflows/workflow.yaml:36]: I have separated out fossa scans already. I dont think anything else in this workflow can be separated into a different file.

**Review by HarrisonWAffel — DISMISSED** [review]

File: .github/workflows/workflow.yaml
Comment (jiaqiluo) [reply .github/workflows/workflow.yaml:25]: I saw the `ci` jon worked [here](https://github.com/mitulshah-suse/rke-tools/actions/runs/9116499897), but I am still wondering how this step helps because dapper downloads git in its [Docker.dapper](https://github.com/rancher/rke-tools/blob/master/Dockerfile.dapper#L6) file instead of using the global git.

**Review by jiaqiluo — APPROVED** [review]
The changes look good to me, but the jobs `build-and-push` and `merge` are not tested anywhere, so we have to wait for the next release to see if they will work.

File: .github/workflows/workflow.yaml
Comment (mitulshah-suse) [reply .github/workflows/workflow.yaml:25]: If we dont have this step, it does not download the `.git` folder and any git commands that we run end up failing with `not a git repository` error. The underlying problem is that for some reason, the checkout needs git binary in the PATH and at a version higher than 2.18, for it to work correctly and download the `.git` folder.

Comment (jiaqiluo) [reply .github/workflows/workflow.yaml:25]: OK, I think I understand the necessity now. Dapper copies or mounts the source code from the runner to its 
 own container to perform operations, if the source code in the runner does not have the .git folder, Dapper will hit the not-a-git-repo error even if it downloads git by itself, therefore, we need to install git before checking out the code in the runner. 

@mitulshah-suse, one more question: why don't you need to run the command `git config --global --add safe.directory $(realpath .)` after installing git per the recommendation in the [GH issue](https://github.com/actions/checkout/issues/363#issuecomment-1163942488)?

Comment (mitulshah-suse) [reply .github/workflows/workflow.yaml:25]: I did not see any permission issue when running it, so did not need to add that step.

**Comment by mitulshah-suse** [comment]
@HarrisonWAffel @jiaqiluo Pushed fossa changes as requested. Please review again.

File: .github/workflows/fossa.yaml
Comment (jiaqiluo) [inline .github/workflows/fossa.yaml:11]: Do not run the workflow on pull requests, because the FOSSA token might be exposed to unauthorized external contributors that open PRs in the repo.

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: .github/workflows/fossa.yaml
Comment (mitulshah-suse) [reply .github/workflows/fossa.yaml]: Updated, thanks for pointing that out.

**Review by jiaqiluo — DISMISSED** [review]

File: scripts/ci
Comment (jakefhyde) [reply scripts/ci:4]: I would still say `$HOME` is not correct. Regardless of the frequency of builds, we need to figure out a way to resolve this without relying on `$HOME`. @mitulshah-suse when you run `make ci` locally, what does `$HOME` resolve to within the dapper container?

**Review by jakefhyde — CHANGES_REQUESTED** [review]

**Review by jiaqiluo — APPROVED** [review]
lgtm

**Review by jakefhyde — APPROVED** [review]

**Pull request merged**
